Android/app/src/androidTest/java/app/intra/util/AppInfoTest.java

```diff
@@ -0,0 +1,57 @@
+package app.intra.util;
+
+import android.os.Parcel;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class AppInfoTest {
+  final String LABEL1 = "label1", LABEL2 = "label2";
+  final String NAME1 = "name.1", NAME2 = "name.2";
+  final AppInfo APP1 = new AppInfo(LABEL1, NAME1);
+  final AppInfo APP2 = new AppInfo(LABEL2, NAME2);
+  final AppInfo NULL1 = new AppInfo(null, NAME1);
+  final AppInfo NULL2 = new AppInfo(null, NAME2);
+
+  @Test
+  public void equal() {
+    assertEquals(0, APP1.compareTo(new AppInfo(LABEL1, NAME1)));
+    assertEquals(0, NULL1.compareTo(new AppInfo(null, NAME1)));
+    assertEquals(0, APP1.compareTo(APP1));
+    assertEquals(0, NULL1.compareTo(NULL1));
+  }
+
+  @Test
+  public void notEqual() {
+    assertEquals(-1, APP1.compareTo(APP2));
+    assertEquals(1, APP2.compareTo(APP1));
+    assertEquals(-1, APP1.compareTo(NULL1));
+    assertEquals(1, NULL1.compareTo(APP1));
+    assertEquals(-1, APP2.compareTo(NULL2));
+    assertEquals(1, NULL2.compareTo(APP2));
+    assertEquals(-1, APP2.compareTo(NULL1));
+    assertEquals(1, NULL1.compareTo(APP2));
+    assertEquals(-1, NULL1.compareTo(NULL2));
+    assertEquals(1, NULL2.compareTo(NULL1));
+  }
+
+  private void roundtrip(AppInfo info) {
```

Comment:
This test method could benefit from a more descriptive name.

Reply:
Done.